### ISSUE TITLE
BO order - No way to display extra carrier information

### ISSUE THREAD

**Issue by Hlavtox** [PrestaShop/PrestaShop#25336]
#### Is your feature request related to a problem?

When creating an order in backoffice, there is no way to for modules to display extra information they can display in FO. For example - selecting a pickup branch, selecting delivery time etc.

#### Describe the solution you'd like

Add a simmilar hooks as in checkout to BO order create.

![select](https://user-images.githubusercontent.com/6097524/125454539-69fd18c6-30de-48ca-903b-cd1ee955fc50.JPG)

**Comment by florine2623** [comment]
Hello @Hlavtox ,

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 

Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know!

Thank you

**Comment by florine2623** [comment]
Ping @PrestaShop/product-team ,

What do you think about @Hlavtox 's suggestion ?

Thanks!

**Comment by fire2** [comment]
YES!! This is my biggest struggle when creating custom modules for shops: there is no hook in the back office "new order" view. I was hoping one would be finally introduced in PS 1.7.7 with the major back office order views migration to Symfony, where a bunch of new hooks were introduced (https://devdocs.prestashop.com/1.7/modules/sample-modules/order-pages-new-hooks/) but unfortunately.... no new hook in the "new order" view. 

I know we can now override the twig template blocks but this is still not a good solution for modules. Since multiple modules might be trying to show additional information in that view. 

My suggestion is to add a hook in /src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/create.html.twig

```
{% block content %}
  <div id="order-creation-container">
    <div class="row">
      <div class="col">
        {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/Create/customer.html.twig' %}
      </div>
    </div>
    <div id="js-cart-info-wrapper">
      <div class="row">
        <div class="col">
          {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/Create/cart.html.twig' %}
        </div>
      </div>
      <div class="row">
        <div class="col">
          {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/Create/cart_rules.html.twig' %}
        </div>
      </div>
      <div class="row">
        <div class="col">
          {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/Create/addresses.html.twig' %}
        </div>
      </div>
      <div class="row">
        <div class="col">
          {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/Create/shipping.html.twig' %}
        </div>
      </div>

          <!-- add hook here for modules to display stuff --> 

      <div class="row">
        <div class="col">
          {% include '@PrestaShop/Admin/Sell/Order/Order/Blocks/Create/summary.html.twig' %}
        </div>
      </div>
    </div>
  </div>
{% endblock %}
```

This should be easy to implement and would save us developers a lot of time and headaches.

**Comment by musaffar-patel** [comment]
In my case, ideally a simple hook in the file below would be very useful:
/src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/Create/shipping.html.twig

**Comment by Hlavtox** [comment]
Still relevant

**Comment by boo-code** [comment]
Still the case on `9.2.x`, and the asymmetry @fire2 described in 2021 is measurable.

The order **view** page carries nine display hooks: `displayAdminOrder`, `displayAdminOrderTop`, `displayAdminOrderMain`, `displayAdminOrderMainBottom`, `displayAdminOrderSide`, `displayAdminOrderSideBottom`, `displayAdminOrderTabLink`, `displayAdminOrderTabContent` and `displayOrderPreview`. That is the set the 1.7.7 Symfony migration brought.

The order **create** page carries exactly one, `displayAdminOrderCreateExtraButtons`, rendered at `Blocks/Create/summary.html.twig:112`. It was added on 2021-01-25 and `hook.xml` describes it as "Add buttons on the create order page dropdown" - a button slot in a dropdown, not a place to render content.

The file @musaffar-patel named, `Blocks/Create/shipping.html.twig`, has no hook at all, so there is still nowhere for a carrier module to put a pickup-point selector or a delivery-time choice next to the carrier it belongs to.

One note for anyone reading the cross-reference: #27937 is closed as completed, but no pull request is linked to it, and I could not find anything in the tree that implements what it asked for. It should not be taken as evidence that this landed.
